src/components/ShowCard.tsx

```diff
@@ -10,25 +10,22 @@ interface MovieCardProps { details: MovieDetailsData | null }
  * @param props | returns details object passed from SearchResultScreen.tsx
  * @returns {JSX.Element} | Single show card
  */
-export default function ShowCard(props: MovieCardProps): JSX.Element {
-	const ratingHandler = (arr: MovieDetailsData): JSX.Element => {
-		const newArr: any[] = [];
-		arr.release_dates.results.filter((item: any) => {
-			if (item.iso_3166_1 === 'US') {
-				newArr.push(item);
+export default function ShowCard(props: MovieCardProps) {
+	const ratingHandler = (arr: MovieDetailsData): JSX.Element | undefined => {
```

Comment:
Any reason we use `undefined` instead of `null`? Since we know if we hit the end of this function we have no rating, it seems like `null` makes more sense.

Reply:
Currently on line 42 the rating handler container is rendered even if the rating is undefined. I would suggest conditionally rendering it based on if rating handler returns a value.

Reply:
@Thenlie `JSX.Element || null` throws TS error `Function lacks ending return statement and return type does not include 'undefined'.`